protos/file.proto

```diff
@@ -78,13 +292,49 @@ message Metadata {
     //   position = stats_page_table[5][0];
     //   length = stats_page_table[5][1];
     // ```
-      uint64 page_table_position = 3;
+    uint64 page_table_position = 3;
   }
 
+  // File statistics.  This field is only present in Lance version 1.
+  //
+  // In Lance version 2 statistics are stored per-column and this field
+  // will not be present.
   StatisticsMetadata statistics = 5;
+
+  // The number of columns in the file (including inner columns when there
+  // are nested fields)
+  //
+  // This can be used to access the column metadata offsets array which is
+  // stored immediately before manifest_position.
+  //
+  // Given N columns the column metadata positions and sizes are stored in a
+  // contiguous buffer of 2*N uint64 values immediately preceding the file
+  // descriptor (or the metadata if the file is not self describing).
+  //
+  // If we let `column_offsets_pos` be:
+  //   manifest_position - (16 * num_columns)
+  //
+  // Then the metadata for column x starts at the uint64:
+  //   file[column_offsets_pos + (16 * x)]
+  // The size of the metadata for column x is given by the uint64:
+  //   file[column_offsets_pos + (16 * x) + 8]
+  //
+  // This field is ignored in Lance version 1 files (the page table is used instead)
+  uint32 num_columns = 6;
+  // The start of the column metadata section
+  //
+  // If column projection is not needed, or if the goal is to cache all column
+  // metadata, then this field can be used to quickly load the entire column metadata
+  // section in a single read without referring to the column metadata offsets array
+  //
+  // This field is ignored in Lance version 1 files
+  uint64 column_metadata_start = 7;
```

Comment:
This gives us the start, how do we know where the end is?

Reply:
I suppose the intention was to have it end at `manifest_position - (16 * num_columns)`